pkg/controller/common/layered_node_state.go

```diff
@@ -39,14 +40,27 @@ func (l *LayeredNodeState) IsDoneAt(mcp *mcfgv1.MachineConfigPool) bool {
 //
 // This augments this check by determining if the desired iamge annotation is
 // equal to what the pool expects.
-func (l *LayeredNodeState) IsUnavailable(mcp *mcfgv1.MachineConfigPool) bool {
-	return isNodeUnavailable(l.node) && l.isDesiredImageEqualToPool(mcp)
+func (l *LayeredNodeState) IsUnavailable(mcp *mcfgv1.MachineConfigPool, layered bool) bool {
```

Comment:
question: Is there an advantage to passing in whether a pool is layered or not instead of inferring it from the pool object?

I ask because when I wrote this, I intended for `LayeredNodeState` to infer whether the pool is layered or not. To be clear: I'm fine with this change, I just want to understand the reason for it :smile:

Reply:
we should only depend on the existence of the MachineOSConfig object as the opt in mechanism to layering. 

I kept the label thing here just for testing